### ISSUE TITLE
Suggested policy: number format with trailing .

### ISSUE THREAD

**Issue by epa** [Perl-Critic/Perl-Critic#622]
You can write the constant 42 as `42` or as `42.` with a trailing dot.  Perlcritic could warn when it sees the latter.

You could extend this to `42.000` and so on, but I suggest that is less useful, since normally the explicit trailing zeroes indicate something about the programmer's intent (perhaps the number is in a table of other numbers which are all to three d.p.).  But the trailing . character just looks like a mistake.

**Comment by Tux** [comment]
I disagree. Adding a trailing dot forces NV instead of IV and is quite useful (and used):
```
$ perl -MDP -we'$a=1;DPeek$a;$b=1.;DPeek$b'
IV(1)
NV(1)
$
```

**Comment by epa** [comment]
Hmm, I suggest that if you want to force a floating point constant better to write 1.0 rather than just 1.

The .0 makes it clear this is intentional, whereas a stray . character is more likely to creep into the code as a typo.  But I don't feel strongly about it.

**Comment by Tux** [comment]
that clarity is within the eye of the beholder. I prefer 1. over 1.0: 1. shows I force float, 1.0 adds specific digits that I did not want to specify.
